Updating contrib guide, making it a bit more practical - WIP (#553)
redhat-documentation/vale-at-red-hat: modules/contributor-guide/partials/proc_extending-styles.adoc

:_module-type: PROCEDURE

[id="proc_extending_styles_{context}"]
= Extending styles in the vale-at-red-hat repository

The vale-at-red-hat repository is an open source project maintained and developed by volunteers. If you want to help developing `vale-red-hat`, feel free to open a PR.

[NOTE]
====
If you use a custom Vale rule set in your Red Hat project, please consider adding it to the vale-at-red-hat repository. 
====

.Prerequisites

* xref:user-guide:installing-vale-cli.adoc[]
* xref:user-guide:understanding-vale-output.adoc[]
* xref:user-guide:defining-a-vale-onboarding-strategy.adoc[]
* You forked and cloned the `vale-at-red-hat` repository using the link:https://www.atlassian.com/git/tutorials/comparing-workflows/forking-workflow[Git forking workflow].

.Procedure

. Configure the rule and the associated test fixtures:
+
* Configure the rule in the `styles` folder: `.vale/styles/<Style>/_<StyleName>_.yml`. 
* Configure the test fixtures in the `fixtures` folder: `.vale/fixtures/<Style>/<StyleName>/testinvalid.adoc|testvalid.adoc`

. Run the `vale` command on the test fixtures to ensure that the rule triggers as you expect, for example:
+
[source,terminal]
----

> $ vale testvalid.adoc

Reviewer: ```suggestion
$ vale .
```

Run against both files :shrug:

----
+
.Example output
+
[source,terminal]
----

> ✔ 0 errors, 0 warnings and 0 suggestions in 1 file.

Reviewer: And show testvalid.adoc and testinvalid.adoc results

:_module-type: PROCEDURE

[id="proc_extending_styles_{context}"]
= Extending styles in the vale-at-red-hat repository

> The vale-at-red-hat repository is an open source project maintained and developed by volunteers. If you want to help developing `vale-red-hat`, feel free to open a PR.

Reviewer 2: The backticks make this look like it might be a specific object or folder. How about:
```suggestion
The vale-at-red-hat repository is an open-source project maintained and developed by volunteers. If you want to contribute to the Vale at Red Hat project, feel free to open a PR.
```

[NOTE]
====
If you use a custom Vale rule set in your Red Hat project, please consider adding it to the vale-at-red-hat repository. 
====

.Prerequisites

* xref:user-guide:installing-vale-cli.adoc[]
* xref:user-guide:understanding-vale-output.adoc[]
* xref:user-guide:defining-a-vale-onboarding-strategy.adoc[]
* You forked and cloned the `vale-at-red-hat` repository using the link:https://www.atlassian.com/git/tutorials/comparing-workflows/forking-workflow[Git forking workflow].

.Procedure

> . Configure the rule and the associated test fixtures:
> +
> * Configure the rule in the `styles` folder: `.vale/styles/<Style>/_<StyleName>_.yml`. 
> * Configure the test fixtures in the `fixtures` folder: `.vale/fixtures/<Style>/<StyleName>/testinvalid.adoc|testvalid.adoc`

Reviewer 2: Maybe simpler and clearer this way. WDYT?
```suggestion
. Configure the rule in the `styles` folder: `.vale/styles/<Style>/_<StyleName>_.yml`. 
. Configure the test fixtures in the `fixtures` folder: `.vale/fixtures/<Style>/<StyleName>/testinvalid.adoc|testvalid.adoc`
```
